### PR TITLE
[LoongArch64] Add LoongArch64 support in libunwind.

### PR DESCRIPTION
**Pull request by shushanhf**
add the libunwind for LoongArch64 alone.

### CODE REVIEW

**Comment by am11** [comment]
@shushanhf, thank you. It looks like it is missing changes in:
https://github.com/dotnet/runtime/blob/3b2a700/src/coreclr/pal/src/libunwind/src/CMakeLists.txt
also please add a line `Apply https://github.com/libunwind/libunwind/pull/316` in version file:
https://github.com/dotnet/runtime/blob/3b2a700/src/coreclr/pal/src/libunwind/libunwind-version.txt

**Comment by shushanhf** [comment]
> @shushanhf, thank you. It looks like it is missing changes in: https://github.com/dotnet/runtime/blob/3b2a700/src/coreclr/pal/src/libunwind/src/CMakeLists.txt also please add a line `Apply https://github.com/libunwind/libunwind/pull/316` in version file: https://github.com/dotnet/runtime/blob/3b2a700/src/coreclr/pal/src/libunwind/libunwind-version.txt

Yes, I had moved it from the #62889.
Request for review again.
Thanks

**Comment by am11** [comment]
@shushanhf, please update `libunwind-version.txt` with upstream patch link which helps us when we bump the library version. Other than that, it looks good. :+1:

cc @janvorli, @jkotas

**Comment by shushanhf** [comment]
> @shushanhf, please update `libunwind-version.txt` with upstream patch link which helps us when we bump the library version. Other than that, it looks good. +1
> 
> cc @janvorli, @jkotas

I am very sorry for I forgot add the version.
Had done.

Thanks for your review again !!!

File: src/coreclr/pal/src/libunwind/libunwind-version.txt [hunk 1]
```diff
@@ -5,3 +5,6 @@ Replace CMakeLists.txt, src/CMakeLists.txt, configure.cmake with .NET custom ver
 Keep .NET oop directory
 Reapply changes from https://github.com/dotnet/runtime/commit/1b5719c2e3dde393531eaeb5b5cde05dabeef5b8
 Apply https://github.com/libunwind/libunwind/pull/317
+
+For LoongArch64:
+Apply https://github.com/libunwind/libunwind/pull/316
```
Comment (kant2002) [inline src/coreclr/pal/src/libunwind/libunwind-version.txt:10]: Also should be added `Apply https://github.com/libunwind/libunwind/pull/322`

Comment (kant2002) [reply src/coreclr/pal/src/libunwind/libunwind-version.txt:10]: At least if I undertand things correctly.

Comment (am11) [reply src/coreclr/pal/src/libunwind/libunwind-version.txt:10]: > Also should be added `Apply https://github.com/libunwind/libunwind/pull/322`

I don't see `atomic_bool` changes in this PR.

**Comment by am11** [comment]
@shushanhf, as agreed previously, everything outside of libunwind directory should go in a separate PR. This PR is about updating libunwind, and not `.*unwind.*` in this repo (which has a vast surface area encompassing PAL, VM, JIT and beyond...). So please revert commits after https://github.com/dotnet/runtime/pull/62979/commits/e9954d44810fe425c676598b69d1a92d20a7c8e9.

**Comment by shushanhf** [comment]
@am11 OK, I will revert it.

File: src/coreclr/pal/src/libunwind/libunwind-version.txt (hunk 1 above)
Comment (shushanhf) [reply src/coreclr/pal/src/libunwind/libunwind-version.txt:10]: I pushed the `atomic_bool` changes to the upstream.
If need, I will update it within this PR.

Comment (shushanhf) [reply src/coreclr/pal/src/libunwind/libunwind-version.txt:10]: merged the `https://github.com/libunwind/libunwind/pull/322` by https://github.com/dotnet/runtime/pull/62979/commits/db234e52bef1fe49cf107535a8fed33bf771d82d

**Comment by shushanhf** [comment]
Close & reopen for trigger CI

**Review by janvorli — APPROVED** [review]
LGTM modulo the comment, thank you!

**Comment by shushanhf** [comment]
> LGTM modulo the comment, thank you!
> @shushanhf can you please add the https://github.com/libunwind/libunwind/pull/322 to the list of "Apply" changes here **too?**

OK,  I will add it.
Thanks

**Comment by shushanhf** [comment]
> Hi @janvorli, could you please take a look so we can merge?

Hi,  @mangod9 
I had updated the patch.
If need, I can update again.
Thanks.

**Comment by janvorli** [comment]
The change looks good, I have already approved it last week with one nit that was fixed since.

**Pull request merged**